linode/manager
# feat: [M3-6749] - Subnet delete dialog

## Description 📝
- add subnet delete dialog to enable deleting of subnets

## Preview 📷

Delete disabled on action menu:
![image](https://github.com/linode/manager/assets/139280159/5efc9c7f-17a4-4c13-ae83-4a9ac7be8e9b)

Subnet Delete Dialog:
![image](https://github.com/linode/manager/assets/139280159/a52e6647-755e-4d7e-b2ec-6998f39e36de)

(tooltip icon color is off in this video recording again -- the color kinda just changes randomly 😭 )
https://github.com/linode/manager/assets/139280159/471dc77b-1975-4b08-bdaa-621df7023022

## How to test 🧪
```
yarn test SubnetDeleteDialog.test.tsx
yarn test VPCSubnetsTable.test.tsx
yarn cy:run -s "cypress/e2e/core/vpc/vpc-details-page.spec.ts"
```
Navigate to a VPC's detail page and:
- Verify that you cannot delete subnets that have linodes assigned to them
- Verify that you can delete subnets that have no linodes assigned to them
